plugins/extra/remind.go

```diff
@@ -200,14 +212,89 @@ func (p *reminderPlugin) InitialDispatch(b *seabird.Bot, m *irc.Message) {
 	go p.remindLoop(b)
 }
 
+// Parses the text string and turns it into a time.Duration
+func (p *reminderPlugin) ParseTime(timeStr string) (time.Duration, error) {
+	buf := bytes.NewBufferString(timeStr)
+	parsed := &timeData{}
+
+	var tmp string
+	var val float64
+	var isFloat bool
+	for buf.Len() > 0 {
+		// Read next character
+		curChar, size, err := buf.ReadRune()
+		if err != nil || size > 1 {
+			return -1, err
+		}
+
+		// Fix uppercase
+		if curChar >= 'A' && curChar <= 'Z' {
+			curChar = unicode.ToLower(curChar)
+		}
+
+		// Parse previous number if we're at a letter
+		if curChar >= 'a' && curChar <= 'z' {
+			if tmp == "" || tmp == "-" || tmp == "." {
+				return -1, errors.New("Blank value passed.")
+			}
+
+			val, err = strconv.ParseFloat(tmp, 64)
+			if err != nil {
+				return -1, err
+			}
+
+			tmp = ""
+		}
+
+		// Store parsed values
+		switch (curChar) {
+		case 'd':
+			parsed.Days = val
+		case 'h':
+			parsed.Hours = val
+		case 'm':
+			parsed.Mins = val
+		case 's':
+			parsed.Secs = val
+		case '.':
+			if !isFloat {
+				isFloat = true
+				tmp += "."
+			} else {
+				return -1, errors.New("Too many decimal points specified.")
+			}
+		default:
+			if curChar >= '0' && curChar <= '9' {
+				tmp += fmt.Sprintf("%c", curChar)
+			} else if tmp == "" && curChar == '-' {
+				// Only allow - at the beginning of numbers
+				tmp += "-"
+			} else {
+				return -1, errors.New("Invalid character specified.")
+			}
+		}
+	}
+
+	// Calculate seconds
+	var nsecs time.Duration
+	nsecs = time.Duration(((parsed.Days * 86400) + (parsed.Hours * 3600) + (parsed.Mins * 60) + parsed.Secs) * 1000000000)
```

Comment:
sorry to be nitpicky here but I think this would be clearer if we used `parsed.Days * Day`, where `Day = time.Hour * 24`. And the same for Hours, Mins, and Secs. That way we also wouldn't need the ` * 1000000000` at the end